### PR TITLE
Get rid of by reference

### LINKED ISSUE #14643: `OC_Util::sanitizeHTML` get rid of by-reference

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes https://github.com/owncloud/core/issues/14643

cc @owncloud/security

### CODE REVIEW

**Comment by LukasReschke** [comment]
Also @nickvergessen due to https://github.com/owncloud/core/pull/14642

File: lib/private/util.php [hunk 1]
```diff
@@ -1177,14 +1177,16 @@ public static function callCheck() {
...
+	 * @param string|array $value
 	 * @return string|array an array of sanitized strings or a single sanitized string, depends on the input parameter.
 	 */
-	public static function sanitizeHTML(&$value) {
+	public static function sanitizeHTML($value) {
 		if (is_array($value)) {
-			array_walk_recursive($value, 'OC_Util::sanitizeHTML');
+			array_walk_recursive($value, function(&$arrayValue) {
+				$arrayValue = self::sanitizeHTML($arrayValue);
+			});
```
Comment (LukasReschke) [inline lib/private/util.php]: For the record: This is unit tested :wink:

**Comment by nickvergessen** [comment]
fine by me

File: lib/private/util.php [hunk 2]
```diff
@@ -1177,14 +1177,16 @@ public static function callCheck() {
...
 	 *
-	 * @param string|array &$value
+	 * @param string|array $value
 	 * @return string|array an array of sanitized strings or a single sanitized string, depends on the input parameter.
 	 */
-	public static function sanitizeHTML(&$value) {
+	public static function sanitizeHTML($value) {
 		if (is_array($value)) {
-			array_walk_recursive($value, 'OC_Util::sanitizeHTML');
+			array_walk_recursive($value, function(&$arrayValue) {
```
Comment (nickvergessen) [inline lib/private/util.php]: could also use array_map here to get rid of this reference?

Comment (LukasReschke) [inline lib/private/util.php]: Done

**Comment by nickvergessen** [comment]
:+1:

File: lib/private/util.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/util.php]: But there is no nested array, to check the recursion :see_no_evil:

Comment (LukasReschke) [inline lib/private/util.php]: Now there is :speak_no_evil:

**Comment by rullzer** [comment]
:+1:

**Pull request merged**